crates/client/src/api.rs

```diff
@@ -172,25 +171,46 @@ impl WithWargHeader for RequestBuilder {
     }
 }
 
+trait WithAuth {
+    fn auth(self, auth_token: &Option<Secret<String>>) -> RequestBuilder;
+}
+
+impl WithAuth for RequestBuilder {
+    fn auth(self, auth_token: &Option<Secret<String>>) -> reqwest::RequestBuilder {
+        if let Some(tok) = auth_token {
+            self.bearer_auth(tok.expose_secret())
+        } else {
+            self
+        }
+    }
+}
+
 /// Represents a Warg API client for communicating with
 /// a Warg registry server.
 pub struct Client {
     url: RegistryUrl,
     client: reqwest::Client,
     warg_registry_header: Option<RegistryDomain>,
+    auth_token: Option<Secret<String>>,
 }
 
 impl Client {
     /// Creates a new API client with the given URL.
-    pub fn new(url: impl IntoUrl) -> Result<Self> {
+    pub fn new(url: impl IntoUrl, auth_token: Option<Secret<String>>) -> Result<Self> {
         let url = RegistryUrl::new(url)?;
         Ok(Self {
             url,
             client: reqwest::Client::new(),
             warg_registry_header: None,
+            auth_token,
```

Comment:
I'd recommend using https://docs.rs/reqwest/0.11.24/reqwest/struct.ClientBuilder.html#method.default_headers to replace all the changes in this file. The first example in the method docs there is basically what you would need (plus prepending `Bearer ` to the value).

Reply:
Think I agree with Calvin here... though if there's something in the middle, the default headers is definitely a smaller change.  Went ahead and pushed the other suggestions.